src/libraries/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -15,10 +16,19 @@ namespace System.Net
     // amounts of data to the server such as from a file stream.
     internal sealed class RequestStream : Stream
     {
-        private readonly MemoryStream _buffer = new MemoryStream();
+        private readonly MemoryStream? _buffer;
+        private readonly StreamBuffer _streamBuffer;
+        private readonly bool _isBuffered;
+        private bool _disposed;
 
-        public RequestStream()
+        public RequestStream(StreamBuffer streamBuffer, bool isBuffered)
         {
+            if (isBuffered)
+            {
+                _buffer = new MemoryStream();
+            }
+            _streamBuffer = streamBuffer;
+            _isBuffered = isBuffered;
```

Comment:
We already have a `StreamBuffer`. Can we skip the different handling here and just always write into it instead of the memory stream?
(should change max capacity if buffering is enabled)